src/psp.ts

```diff
@@ -0,0 +1,502 @@
+                  type: 'address',
+                },
+              ],
+              name: 'isAuthorized',
+              outputs: [
+                {
+                  internalType: 'bool',
+                  name: '',
+                  type: 'bool',
+                },
+              ],
+              stateMutability: 'view',
+              type: 'function',
+            },
+          ];
+          const authorizesRequesterForAirnodeEndpoint = async (authorizerAddress: string): Promise<any> => {
+            const authorizer = new ethers.Contract(authorizerAddress, authorizerAbi, provider);
+            return await authorizer.isAuthorized(airnodeAddress, endpointId, subscription.requester);
+          };
+          const authorized =
+            isEmpty(chain.authorizers) || chain.authorizers.some(authorizesRequesterForAirnodeEndpoint);
+          if (!authorized) {
+            node.logger.info('Requester not authorized', subscriptionIdLogOptions);
+            continue;
+          }
+
+          // **************************************************************************
+          // Make API call
+          // **************************************************************************
+          node.logger.debug('Making API requests...', subscriptionIdLogOptions);
+
+          const [error, logsData] = await retryGo(() =>
+            readApiValue({
+              airnodeAddress,
+              oises,
+              apiCredentials,
+              id: subscriptionId,
+              trigger: subscription,
+            })
+          );
+          if (!isNil(error) || isNil(logsData)) {
+            node.logger.warn('Failed to fecth API value', subscriptionIdLogOptions);
+            continue;
+          }
+          const [logs, data] = logsData;
+          node.logger.logPending(logs, subscriptionIdLogOptions);
+
+          const apiValue = data[subscriptionId];
+          if (isNil(apiValue)) {
+            node.logger.warn('API value is missing. Skipping update...', subscriptionIdLogOptions);
+            continue;
+          }
+
+          // **************************************************************************
+          // Check conditions
+          // **************************************************************************
+          node.logger.debug('Checking conditions...', subscriptionIdLogOptions);
+
+          // Is this step really needed?
+          // https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/contracts/dapis/DapiServer.sol#L20-L23
+
+          const encodedFulfillmentData = ethers.utils.defaultAbiCoder.encode(['int256'], [apiValue]);
+
+          // TODO: Should "subscription.conditions" be already encoded in config?
+          // const encodedConditionParameters = ethers.utils.defaultAbiCoder.encode(
+          //   ['uint256'],
+          //   [subscription.conditions] // How is this supposed to be set in config? i.e. (10**18 / 10) for 10%?
+          // );
```

Comment:
Yeah, I think you should decode `conditions` using airnode-abi each time you need to do the condition check to get `_conditionParameters`

Reply:
To keep it simple at first I just register the subscription with `conditions` being the threshold value but I guess this can become more complicated and `airnode-abi` will make more sense to be used.

Reply:
That would definitely work in this specific case but proper airnode-abi encoding will allow us to use proto-psp with contracts other than DapiServer